collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,143 @@
-class ComparableMixin(object):
+from copy import deepcopy
+
+class ComparableMixin(object): # Subclassed in List & Dict
     def __eq__(self, other):
-        pass
+        data = getattr(
+            self,
+            self.DATA_ATTR_NAME
+        )
+        data2 = getattr(
+            other,
+            other.DATA_ATTR_NAME
+        )
+        return data == data2
 
     def __ne__(self, other):
         return not self == other
 
```

Comment:
I've been going over solutions from other groups to get a sense of where we went wrong. This was an elegant way to get `__ne__` that's really helpful for me in terms of understanding the magic methods and the ways that they operate. Since you've already defined `__eq__` verbosely, it's going to be called by the `==` operator in `__ne__`.